cumulus_library/actions/builder.py

```diff
@@ -136,13 +158,31 @@ def run_protected_table_builder(
 
 
 def _run_workflow(
-    config: base_utils.StudyConfig, manifest: study_manifest.StudyManifest, filename: str
-) -> None:
+    config: base_utils.StudyConfig,
+    manifest: study_manifest.StudyManifest,
+    filename: str,
+    prepare: str,
+    data_path: pathlib.Path,
+    query_count: int,
+) -> int:
     """Loads workflow config from toml definitions and executes workflow
 
     :param config: a StudyConfig object
     :param manifest: a StudyManifest object
+    :param filename: Filename of the workflow config
+    :param prepare: If true, will render query instead of executing
+    :param data_path: If prepare is true, the path to write rendered data to
+    :param query_count: if prepare is true, the number of queries already rendered
+    :returns: a count of processed queries
     """
+    toml_path = pathlib.Path(f"{manifest._study_path}/{filename}")
+    if prepare:
+        with open(toml_path) as file:
+            workflow_config = file.read()
+        _render_output(
+            manifest.get_study_prefix(), [workflow_config], data_path, filename, query_count, True
```

Comment:
nit: I hate to see a bare True/False - can you make that parameter kwarg only? (using `*` in the param list).

Reply:
i skipped it since it was in module and internal, but for you? sure.